bundles/org.openhab.binding.zway/src/main/java/org/openhab/binding/zway/internal/discovery/ZWayBridgeDiscoveryService.java

```diff
@@ -56,36 +54,11 @@ public ZWayBridgeDiscoveryService() {
     private void scan() {
         logger.debug("Starting scan for Z-Way Server");
 
-        ValidateIPV4 validator = new ValidateIPV4();
+        List<InetAddress> addressesToScan = NetUtil.getFullRangeOfAddressesToScan();
+        logger.debug("Performing discovery on {} ip addresses", addressesToScan.size());
 
-        try {
-            Enumeration<NetworkInterface> enumNetworkInterface = NetworkInterface.getNetworkInterfaces();
-            while (enumNetworkInterface.hasMoreElements()) {
-                NetworkInterface networkInterface = enumNetworkInterface.nextElement();
-                if (networkInterface.isUp() && !networkInterface.isVirtual() && !networkInterface.isLoopback()) {
-                    for (InterfaceAddress address : networkInterface.getInterfaceAddresses()) {
-                        if (validator.isValidIPV4(address.getAddress().getHostAddress())) {
-                            String ipAddress = address.getAddress().getHostAddress();
-                            Short prefix = address.getNetworkPrefixLength();
-
-                            logger.debug("Scan IP address for Z-Way Server: {}", ipAddress);
-
-                            // Search on localhost first
-                            scheduler.execute(new ZWayServerScan(ipAddress));
-
-                            String subnet = ipAddress + "/" + prefix;
-                            SubnetUtils utils = new SubnetUtils(subnet);
-                            String[] addresses = utils.getInfo().getAllAddresses();
-
-                            for (String addressInSubnet : addresses) {
-                                scheduler.execute(new ZWayServerScan(addressInSubnet));
-                            }
-                        }
-                    }
-                }
-            }
-        } catch (SocketException e) {
-            logger.warn("Error occurred while searching Z-Way servers ({})", e.getMessage());
+        for (final InetAddress address : addressesToScan) {
```

Comment:
Previous code was ignoring loopback interface, any interface that is not up and any virtual interface.
New code does not ?

Previous code also first scanning with the interface main IP before scanning with all the other IPs of this interface.
Does the new code do the same. If not, is it not important ?

Reply:
Starting with the interface ip can be a minor improvement. It only affects users that have the z-way software installed on the same server and interface as openHAB. The result will be a little bit faster in the inbox, neverteless the scanning proceeds and all servers will be found. I considered this as an acceptable trade off.

By default the NetUtils does not filter for virtual interfaces, and i could not think of a reason why you would want to filter on this. It would prevent one to make use of vlans, docker or other advanced setups with this binding. 

`NetUtil.getFullRangeOfAddressesToScan()`  calls `NetUtil.getAllInterfaceAddresses()` and that method filters for loopback and if it's up:

https://github.com/openhab/openhab-core/blob/389f6a34343a723442829b7d9841240adc2f2a9c/bundles/org.openhab.core/src/main/java/org/openhab/core/net/NetUtil.java#L344

Reply:
Gentle ping @lolodomo

Reply:
Ok